overlord/configstate/configcore/services.go

```diff
@@ -37,22 +40,36 @@ func (l *sysdLogger) Notify(status string) {
 // where "true" means disabled and "false" means enabled.
 func switchDisableService(serviceName, value string) error {
 	sysd := systemd.New(dirs.GlobalRootDir, &sysdLogger{})
+	sshCanary := filepath.Join(dirs.GlobalRootDir, "/etc/ssh/sshd_not_to_be_run")
 
 	switch value {
 	case "true":
-		if err := sysd.Disable(serviceName); err != nil {
-			return err
-		}
-		if err := sysd.Mask(serviceName); err != nil {
-			return err
+		if serviceName == "ssh.service" {
```

Comment:
Thanks for finding a workaround on short notice, Zygmunt.

I heard you've made this work in practice, which is great, but I'm still a bit lost here. It's not clear to me what this is doing yet, and how it makes sense to error saying that snapd configuration has it a given way when this is in fact the snapd configuration.

Let's please discuss this tomorrow.